cmd/img_mgmt/src/img_mgmt.c

```diff
@@ -394,42 +394,49 @@ img_mgmt_upload(struct mgmt_ctxt *ctxt)
         .data_len = 0,
         .data_sha_len = 0,
         .upgrade = false,
+        .image = 0,
     };
 
     const struct cbor_attr_t off_attr[] = {
         [0] = {
+            .attribute = "image",
+            .type = CborAttrUnsignedIntegerType,
+            .addr.integer = &req.image,
```

Comment:
```
Error: repos/apache-mynewt-mcumgr/cmd/img_mgmt/src/img_mgmt.c: In function 'img_mgmt_upload':
repos/apache-mynewt-mcumgr/cmd/img_mgmt/src/img_mgmt.c:404:29: error: pointer targets in initialization of 'long long int *' from 'long long unsigned int *' differ in signedness [-Werror=pointer-sign]
             .addr.integer = &req.image,
                             ^
repos/apache-mynewt-mcumgr/cmd/img_mgmt/src/img_mgmt.c:404:29: note: (near initialization for 'off_attr[0].addr.integer')
```

When building I get this error, since you chose the type to be `CborAttrUnsignedIntegerType`, this line should read `.addr.uinteger = &req.image`, what do you think?

Otherwise the PR looks good.

Reply:
> (...) this line should read `.addr.uinteger = &req.image`, what do you think?

Yes, you are right this should be fixed.

Reply:
I have also changed the comment on the `image` member of the `img_mgmt_upload_req` as the "auto-select" has been system-specific.